pymc3/distributions/multivariate.py

```diff
@@ -255,56 +262,53 @@ def random(self, point=None, size=None):
         -------
         array
         """
-        if size is None:
-            size = tuple()
+        size = to_tuple(size)
+
+        param_attribute = "chol_cov" if self._cov_type == "chol" else self._cov_type
+        mu, param = draw_values([self.mu, getattr(self, param_attribute)], point=point, size=size)
+        if tuple(self.shape):
+            dist_shape = tuple(self.shape)
         else:
-            if not isinstance(size, tuple):
-                try:
-                    size = tuple(size)
-                except TypeError:
-                    size = (size,)
+            if is_fast_drawable(self.mu) or point:
+                batch_shape = mu.shape[:-1]
```

Comment:
You cannot depend only on `mu` to infer a batch shape. You also have to consider the covariance matrix.

Reply:
I thought a lot on this to not consider covariance matrix to determine batch_shape because PyMC3 does not allow to form covariance matrices with `ndim>2`. So, the only two dimensions we have in covariance matrix, correspond to event shape (k * k as in TFP). 

The only possibility for batch shape would be through `mu`. 
Feel free to correct me wherever I go wrong.

Reply:
You can have a random variable like Wishart generate `sample_shape + event_shape` arrays representing independent draws from random matrices

Reply:
Oh @lucianopaz . I think I get the idea. We cannot have batches for covariance matrices, but we can have `sample_shape`. So, we only need to insert batch dimensions which we get only from `mu`, in the middle of samples (between sample shape and event shape) drawn from either Wishart or LKJ distribution. Thoughts on this?

Reply:
Now, the batch shapes are inserted in between `sample_shape` and `event_shape` for parametrizations in de1fa05 commit.